### ISSUE TITLE
Unexplained base content of sequence alignment generated under UNREST model

### ISSUE THREAD

**Issue by kurohuji** [iqtree/iqtree2#160]
I am trying to generate sequence alignments under the UNREST model. However, I am puzzled by the base content of the generated sequences.
I run the following command
```
iqtree2 --alisim alignment_vwxy-1000_unr-31331333333-25252525 -t vwxy_1000.nwk -m UNREST{3.0/1.0/3.0/3.0/1.0/3.0/3.0/3.0/3.0/3.0/3.0}+F{0.25/0.25/0.25/0.25} --length 25000 -redo
```
```
((V:1.0,W:1.0):0.0,(X:1.0,Y:1.0):0.0); # vwxy_1000.nwk
```
I expected that the generated sequence containing equal amounts of A, T, C and only a small amount of G. In fact, the sequence contained small amount of G, but a larger amount of T than A or C.

| Base | Content (%) |
| ---- | ---- |
| A | 28.6 |
| C | 27.7 |
| G | 11.3 |
| T | 32.3 |

In addition, I ran three commands to verify the behavior of the UNREST model.
```
iqtree2 --alisim alignment_vwxy-1000_jc -t vwxy_1000.nwk -m JC --length 25000 -redo
iqtree2 --alisim alignment_vwxy-1000_gtr-11111-25252525 -t vwxy_1000.nwk -m GTR{1.0/1.0/1.0/1.0/1.0}+F{0.25/0.25/0.25/0.25} --length 25000 -redo
iqtree2 --alisim alignment_vwxy-1000_unr-11111111111-25252525 -t vwxy_1000.nwk -m UNREST{1.0/1.0/1.0/1.0/1.0/1.0/1.0/1.0/1.0/1.0/1.0}+F{0.25/0.25/0.25/0.25} --length 25000 -redo
```
My understanding is that these commands should be equivalent. However, only the UNREST model showed a base content that deviated significantly from the expected value.
|  | JC | GTR | UNREST |
| ---- | ---- | ---- | ---- |
| A | 25.1 | 25.2 | 28.0 |
| C | 25.1 | 24.7 | 24.1 |
| G | 24.9 | 25.0 | 22.2 |
| T | 24.9 | 25.2 | 25.6 |

**Comment by bqminh** [comment]
Thanks, that's interesting observation. 
Right now I can only tell you that: UNREST model has nucleotide frequencies intrinsically in itself, i.e., +F{...} has no effect on the model. Rather, the frequencies are inferred directly from the rate matrix.
But I'm also puzzled by the last experiment you did, i.e., UNREST with exchange rates of 1.0 everywhere. I'd also expect the frequencies to be 0.25.
@trongnhanuit may be able to help further.

**Comment by roblanf** [comment]
I was intrigued by this so I did a little experiment. I ran your three commands (JC, GTR, UNREST) 100 times each, and recorded the mean, SD, min, and max of the base frequencies.

Here are the results (mean is first, sd, min, max are in brackets):

**JC model**
A: 0.2500(0.0014, 0.2465, 0.2543)
C: 0.2502(0.0015, 0.2457, 0.2544)
G: 0.2499(0.0015, 0.2458, 0.2544)
T: 0.2498(0.0014, 0.2446, 0.2535)

**GTR model**
A: 0.2500(0.0014, 0.2459, 0.2541)
C: 0.2501(0.0014, 0.2456, 0.2547)
G: 0.2500(0.0015, 0.2456, 0.2533)
T: 0.2499(0.0015, 0.2460, 0.2536)

**UNREST model**
A: 0.2602(0.0148, 0.2066, 0.3197)
C: 0.2415(0.0128, 0.2183, 0.2862)
G: 0.2463(0.0150, 0.2093, 0.2939)
T: 0.2520(0.0156, 0.2113, 0.2830)

So, it does seem to me that the SDs are ~10x higher for the UNREST model. 

The question (and I don't know the answer offhand, but maybe @bqminh and @trongnhanuit might help): is this expected? Is this something inherent in a more parameterised model? To me it seems like it shouldn't be different, given the parameterisations and values given. I.e. it seems like this UNREST should be equivalent to the GTR.

**Comment by roblanf** [comment]
Doing the same for your first command should allay your fears a bit though. Over 250 replicates there is no bias towards T.

I.e. if I run your first command where you expect less G, but roughly equal of the others, I get this:

**UNREST**
A: 0.2971(0.0133, 0.2495, 0.3384)
C: 0.2792(0.0117, 0.2497, 0.3223)
G: 0.1337(0.0148, 0.0979, 0.1812)
T: 0.2900(0.0146, 0.2440, 0.3311)

In other words, it's not unusual to get A, C, or T proportions anywhere from ~25% to ~33%. In this sense, the example you show just seems to be a reasonably extreme one, but not surprising in the statistical sense @kurohuji 

The question above remains though - should UNREST have 10x the SD of GTR?

**Comment by trongnhanuit** [comment]
Hi all,
Thank you @bqminh and @roblanf for testings.

This is a bug, which has been fixed in the recent release: [Version 2.3.5.cmaple](https://github.com/iqtree/iqtree2/releases/tag/v2.3.5.cmaple). 

Thank you @kurohuji for using AliSim and reporting the bug. I'll close this bug report now. Please feel free to contact us if you have any question when using AliSim. 

Thanks,
Nhan
